internal/controller/http/command.go

```diff
@@ -64,15 +73,18 @@ func (c *RestController) Command(writer http.ResponseWriter, request *http.Reque
 
 	// return event in http response if specified (default yes)
 	if ok, exist := reserved[v2.ReturnEvent]; !exist || ok[0] == v2.ValueYes {
-		// TODO: the usage of CBOR encoding for binary reading is under discussion
 		c.sendResponse(writer, request, v2.ApiDeviceNameCommandNameRoute, res, http.StatusOK)
 	}
 }
 
-func readBodyAsString(req *http.Request) (string, errors.EdgeX) {
+func readBodyAsString(req *http.Request, maxRequestSize int64) (string, errors.EdgeX) {
 	defer req.Body.Close()
 	body, err := ioutil.ReadAll(req.Body)
 	if err != nil {
+		if len(body) >= int(maxRequestSize) {
```

Comment:
Will `http.MaxBytesReader` return an error if the body size exceeds max amount? If it does return an error isn't `body` undefined? Should be checking the error returned rather than the `len(body)`

Reply:
Also, what if maxRequestSize is `0` ?

Reply:
No, `http.MaxBytesReader` returned `maxByteReader` struct which wraps original `request.Body` and other additional metadata(e.g. how many bytes remaining). And the error only returned when the body is actually being read.

Maybe we can check if the error is specifically threw by `maxByteReader` instead of checking the `len(body)`, for reference: https://github.com/golang/go/issues/41493#issuecomment-722996128, which is more clear/understandable at this point.